### PR TITLE
Automated Updates: only append dependencies with property to original requirements

### PR DESCRIPTION
**Pull request by cuixq**
Maven original requirements are used to look up if the original requirement contains a property or not, so we don't need all original requirements but only the ones with properties.

### CODE REVIEW

File: internal/resolution/manifest/maven.go [hunk 1]
```diff
@@ -24,8 +24,8 @@ const (
 )
 
 type MavenManifestSpecific struct {
-	Properties      []PropertyWithOrigin
-	OriginalImports []resolve.RequirementVersion
+	Properties           []PropertyWithOrigin
+	OriginalRequirements []resolve.RequirementVersion
```
Comment (oliverchang) [inline internal/resolution/manifest/maven.go:28]: Should we add a  comment or name this variable a bit better to capture the fact that these are requirements with properties only?

RequirementsWithProperties?

**Review by michaelkedar — APPROVED** [review]

File: internal/resolution/manifest/maven.go (hunk 1 above)
Comment (cuixq) [reply internal/resolution/manifest/maven.go]: done!

**Review by oliverchang — APPROVED** [review]

File: internal/resolution/manifest/maven.go [hunk 2]
```diff
@@ -139,15 +131,40 @@ func (m MavenManifestIO) Read(df lockfile.DepFile) (Manifest, error) {
...
-			Properties:      properties,
-			OriginalImports: originalImports,
+			Properties:                 properties,
+			RequirementsWithProperties: reqsWithProps,
 		},
 	}, nil
 }
 
+// ContainsProperty returns whether a string contains property placeholder.
+func ContainsProperty(str maven.String) bool {
```
Comment (another-rex) [inline internal/resolution/manifest/maven.go:144]: Add some examples of property placeholders in the docstring (or add a unit test for it.)

**Review by another-rex — APPROVED** [review]
LGTM!

**Pull request merged**